2015-Fall/oPolo/Calculator/Calculator/CalculatorBrain.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  CalculatorBrain.swift
+//  Calculator
+//
+//  Created by Jonas Bruun Jacobsen on 15/09/15.
+//  Copyright (c) 2015 Jonas Bruun Jacobsen. All rights reserved.
+//
+
+import Foundation
+
+class CalculatorBrain {
+    private enum Op: Printable {
+        case Operand(Double)
+        case Constant(String, Double)
+        case UnaryOperation(String, Double -> Double)
+        case BinaryOperation(String, (Double, Double) -> Double)
+        
+        var description: String {
+            get {
+                switch self {
+                case .Operand(let operand):
+                    return "\(operand)"
+                case .Constant(let constant, _):
+                    return constant
+                case .UnaryOperation(let symbol, _):
+                    return symbol
+                case .BinaryOperation(let symbol, _):
+                    return symbol
+                }
+            }
+        }
+    }
+    
+    private var opStack = [Op]() {
+        didSet {
+            //Fires a notification when a new operand/operator/constant/etc is added to the stack.
+            NSNotificationCenter.defaultCenter().postNotificationName("QueuedCalculationsChanged", object: nil)
```

Comment:
I'm not sure what you're trying to achieve by this. Are you going for the ability to listen for new operands in other places of the app?